packages/portfolio-contract/tools/plan-solve.ts

```diff
@@ -272,103 +268,56 @@ const prettyJsonable = (obj: unknown): string => {
-        attrParts[0] = 'thru';
-      } else if (attrParts[0] !== 'allow') {
-        // Replace everything after the first "_" with a unique counter.
-        attrParts = [
-          attrParts[0],
-          provideLazyMap(mappedAttrs, attr, () => mappedAttrs.size + 1),
-        ];
-      }
-      const prefix = attrParts.join('_');
-      const lines = [`\\# ${attr} ${prettyJsonable(constraint)}`];
-      for (const [opName, value] of Object.entries(constraint)) {
-        if (!Number.isFinite(value)) continue;
-        const label = `${prefix}_${opName.slice(0, 3)}`;
-        const op = cplexLpOpForName[opName];
-        lines.push(`${label}: ${makeSumExpr(attr)} ${op} ${value}`);
-      }
-      return lines;
-    })
-    .join('\n  ')}
-BINARY
-  ${Object.keys(model.binaries).join(' ')}
-GENERAL
-  ${Object.keys(model.ints).join(' ')}
-END`.trim();
-  return `${text}\n`;
-};
-
 // This operation is async to allow future use of async solvers if needed
 export const solveRebalance = async (
   model: LpModel,
   graph: RebalanceGraph,
 ): Promise<{ flows: SolvedEdgeFlow[]; detail?: Record<string, unknown> }> => {
-  const cplexLpText = toCplexLpText(model);
-  const matrixResult = await Promise.resolve(highsSolverP)
-    .then(solver => solver.solve(cplexLpText))
-    .catch(error => ({ Status: undefined, error }));
-  if (matrixResult.Status !== 'Optimal') {
-    // Switch to javascript-lp-solver for diagnostics.
-    const jsResult = jsLPSolver.Solve(model, 1e-9);
+  await null;
+  const jsResult = jsLPSolver.Solve(model, 1e-9);
+
+  // jsLPSolver returns an object with variable values
+  // The 'feasible' flag can be overly strict, so we check if we got a result
+  // instead. If result is undefined or there are no variable values, it's truly infeasible.
+  if (
+    !jsResult ||
+    (jsResult.feasible === false && jsResult.result === undefined)
+  ) {
     if (graph.debug) {
       // Emit richer context only on demand to avoid noisy passing runs
-      const { error } = matrixResult as any;
-      if (error) {
-        console.error('[CPLEX LP solver] Error:', error.message, cplexLpText);
-        throw Fail`Invalid CPLEX LP model: ${error.message}`;
-      }
       let msg = formatInfeasibleDiagnostics(graph, model);
       msg += ` | ${prettyJsonable(jsResult)}`;
       console.error('[solver] No feasible solution. Diagnostics:', msg);
       throw Fail`No feasible solution: ${msg}`;
     }
     throw Fail`No feasible solution: ${jsResult}`;
   }
+
   const flows: SolvedEdgeFlow[] = [];
   for (const edge of graph.edges) {
     const { id } = edge;
     const flowKey = `via_${id}`;
-    const flow = matrixResult.Columns[flowKey]?.Primal;
-    const used = (flow ?? 0) > FLOW_EPS;
+    const rawFlow = jsResult[flowKey];
+    // jsLPSolver returns floating-point values, round to nearest integer
+    const flow = rawFlow ? Math.round(rawFlow) : 0;
+    const used = flow > FLOW_EPS;
     if (used) flows.push({ edge, flow, used: true });
```

Comment:
The output from `Math.round` will only be ≤ FLOW_EPS when it is zero.
```suggestion
    // jsLPSolver returns floating-point values, round to nearest integer
    const flow = rawFlow ? Math.round(rawFlow) : 0;
    if (flow !== 0) flows.push({ edge, flow, used: true });
```